src/hill_climber.rs

```diff
@@ -0,0 +1,108 @@
+use core::slice;
+use std::marker::PhantomData;
+
+use bon::Builder;
+use ec_core::{individual::scorer::Scorer, operator::mutator::Mutator};
+use itertools::Itertools;
+use rand::{prelude::Distribution, rng};
+
+#[derive(Debug, thiserror::Error)]
+pub enum HillClimberError<MutationError> {
+    #[error(transparent)]
+    Mutation(#[from] MutationError),
+
+    #[error("Processed zero-sized chunk")]
+    ZeroSizedChunk,
+}
+
+#[derive(Debug, Builder)]
+pub struct HillClimber<Ge, GM, Mut, Sc, Scr, Ins>
+// You typically wouldn't put all these constraints on the struct itself, instead
+// you'd put them on the `impl` block for the struct. But I'm doing it here to
+// make the constraints more visible and (hopefully) make some of the error
+// messages more helpful for people new to Rust.
+where
+    Ge: Clone + std::fmt::Debug + Sync + Send,
```

Comment:
Do we still use the debug bound?

Reply:
Nope we didn't. I'll remove that.

Actually, if we're not going to support parallelism here, we presumably don't need all this `Send + Sync` either. I'll remove that as well.

Reply:
I can't resolve this conversation (since I only have read access to this repo I think) so feel free to resolve it.